HeterogeneousCore/SonicTriton/src/TritonService.cc

```diff
@@ -0,0 +1,293 @@
+#include "HeterogeneousCore/SonicTriton/interface/TritonService.h"
+#include "HeterogeneousCore/SonicTriton/interface/triton_utils.h"
+
+#include "DataFormats/Provenance/interface/ModuleDescription.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+#include "FWCore/ServiceRegistry/interface/ActivityRegistry.h"
+#include "FWCore/ServiceRegistry/interface/ProcessContext.h"
+#include "FWCore/Utilities/interface/Exception.h"
+
+#include "grpc_client.h"
+#include "grpc_service.pb.h"
+
+#include <cstdio>
+#include <cstdlib>
+#include <filesystem>
+#include <utility>
+#include <tuple>
+
+namespace ni = nvidia::inferenceserver;
+namespace nic = ni::client;
+
+const std::string TritonService::Server::fallbackName{"fallback"};
+const std::string TritonService::Server::fallbackUrl{"0.0.0.0:8001"};
+
+namespace {
+  std::pair<std::string, int> execSys(const std::string& cmd) {
+    //redirect stderr to stdout
+    auto pipe = popen((cmd + " 2>&1").c_str(), "r");
```

Comment:
I wonder if the shell is always guaranteed to be sh-compatible (IIRC csh uses different syntax to redirect the stderr).

@Dr15Jones Could you review this function?

Reply:
I admit to not being an expert on this stuff. I had looked it up when I was writing the SharedMemory package but I no longer remember the details. @dan131riley suggestions?

Reply:
From the `popen` documentation: https://pubs.opengroup.org/onlinepubs/009695399/functions/popen.html it says that it always uses the `sh` shell to process the command.

Reply:
Yes, `popen` is defined to use `/bin/sh -c`, and `2>&1` is POSIX-compliant for redirecting stderr to stdin.

Reply:
I based this on https://github.com/cms-sw/cmssw/blob/master/CommonTools/Utils/src/popenCPP.cc. (I didn't reuse that directly because a) it seems that it's not intended to be used outside of that package (header file in src dir) and b) it doesn't keep track of error codes.)